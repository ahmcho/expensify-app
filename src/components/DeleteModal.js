import React from 'react';
import Modal from 'react-modal';

const DeleteModal = (props) => (
	<Modal
		isOpen={!!props.isModalOpen}
		onRequestClose={props.handleDeleteSelectedOption}
		contentLabel="Delete modal"
		ariaHideApp={false}
		closeTimeoutMS={200}
		className="modal"
	>
		<h3 className="modal__title">Are you sure you want to delete this expense?</h3>
		<button className="button modal__button" onClick={props.handleCloseAndDelete}>Yes</button>	
        <button className="button modal__button" onClick={props.handleCloseModalAndNotDelete}>No</button>
	</Modal>
)

export default DeleteModal;