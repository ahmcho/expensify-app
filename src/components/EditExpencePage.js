import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses';
import { removeExpense } from '../actions/expenses';

export class EditExpencePage extends React.Component{
	onSubmit = (expense)=>{
		this.props.editExpense(this.props.expense.id, expense);
		this.props.history.push('/');
		console.log('updated!', expense);
	}
	onRemove = () => {
		this.props.removeExpense({id: this.props.expense.id});
		this.props.history.push('/');
	}
	render(){
		return(
			<div>
				<ExpenseForm
					expense={this.props.expense} 
					onSubmit={this.onSubmit}
				/>
				<button onClick={this.onRemove}>Remove</button>
			</div>
		)
	}
}

const mapDispatchToProps = (dispatch,props) => ({
	editExpense: (id, expense) => dispatch(editExpense(id,expense)),
	removeExpense: (data) => dispatch(removeExpense(data))
})

const mapStateToProps = (state, props) => ({ expense: state.expenses.find((expense)=> expense.id === props.match.params.id) });

export default connect(mapStateToProps,mapDispatchToProps)(EditExpencePage);