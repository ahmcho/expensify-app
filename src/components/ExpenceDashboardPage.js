import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from  './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenceDashboardPage = () => (
	<div>
		<ExpensesSummary />
		<ExpenseListFilters />
		<ExpenseList />
	</div>
)



export default ExpenceDashboardPage;