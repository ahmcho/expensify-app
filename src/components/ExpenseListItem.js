import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
const ExpenseListItem = ({  id, description, amount, createdAt,note }) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h3>Description: {description}</h3>
		</Link>
		<p>Amount: {numeral(amount / 100).format('$0,0.00')} </p>
		<p>Created at: {moment(createdAt).format('MMMM Do, YYYY')} </p>
		<p>Note: {note}</p>

	</div>
);

export default ExpenseListItem;