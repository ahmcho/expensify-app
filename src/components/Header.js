import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
	<header>
		<h1>Expencify</h1>
		<ul>
			<li><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></li>
			<li><NavLink to="/create"  activeClassName="is-active">Create Expense</NavLink></li>
			<li><button onClick={startLogout}>Logout</button></li>
		</ul>
	</header>
)

const mapDispatchToProps = (dispatch) => ({
	startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);