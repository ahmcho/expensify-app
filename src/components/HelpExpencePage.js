import React from 'react';

const HelpExpencePage = () => (
	<div>
		This is from my help expence page
	</div>
)

export default HelpExpencePage;