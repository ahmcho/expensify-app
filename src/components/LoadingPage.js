import React from 'react';
import { LoginPage } from './LoginPage';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/loader.gif"></img>
    </div>
)

export default LoadingPage;