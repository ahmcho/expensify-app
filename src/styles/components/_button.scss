.button{
    border:none;
    color: white;
    font-size: $font-size-large;
    font-weight: 300;
    background-color: $blue;
    display: inline-block;
    padding: $s-size;
    text-decoration: none;
    line-height: 1;
}

.button--link{
    background: none;
}
.button--secondary{
    background-color: #888888;
}