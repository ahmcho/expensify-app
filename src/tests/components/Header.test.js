import { shallow } from 'enzyme';
import React from 'react';
import {Header} from '../../components/Header';

test('Should render header correctly', () => {
    const wrapper = shallow(<Header startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout on button click', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
})