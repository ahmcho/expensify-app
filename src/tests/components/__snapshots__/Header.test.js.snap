// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<header>
  <h1>
    Expencify
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        to="/create"
      >
        Create Expense
      </NavLink>
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </li>
  </ul>
</header>
`;
